src/mds/MDCache.cc

```diff
@@ -2166,7 +2166,8 @@ void MDCache::predirty_journal_parents(MutationRef mut, EMetaBlob *blob,
     // rstat
     if (!primary_dn) {
       // don't update parent this pass
-    } else if (!linkunlink && !(pin->nestlock.can_wrlock(-1) &&
+    } else if (pin->is_auth() && !linkunlink && 
+				!(pin->nestlock.can_wrlock(-1) &&
 				pin->versionlock.can_wrlock())) {
```

Comment:
Actually, it's a scatter-gather bug. dirtyrstat inode isn't processed properly when it and its parent inode belong to different auth mds. CInode::finish_scatter_update() should call CDir::assimilate_dirty_rstat_inodes() when inode is not auth and next lock state is MIX.